src/common/readfield_nc.f90

```diff
@@ -658,9 +660,186 @@ subroutine read_precipitation(ncid, nhdiff, timepos, timeposm1)
+
+      where (cloud_water < 0.0)
+        cloud_water = 0.0
+      end where
+      where (cloud_ice < 0.0)
+        cloud_ice = 0.0
+      end where
+      cw3d(:,:,k) = cloud_water + cloud_ice
+      cw3d(:,:,k) = cw3d(:,:,k) * pdiff / g
+
+      call nfcheckload(ncid, met_params%cloud_fraction, start, count, cloud_cover(:,:,k))
+    enddo
+  end subroutine
+
+  !> Read and convert fields from ecemep input to what we need for 3D precip
+  subroutine read_extra_precipitation_fields_infer_3d_precip(ncid, timepos)
+    use iso_fortran_env, only: error_unit, real64
+    use snaptabML, only: g
+    use snapfldML, only: ps2, precip3d, cw3d, cloud_cover, enspos, precip
+    use snapgrdML, only: ahalf, bhalf, klevel
+    use snapdimML, only: nx, ny, nk
+    use snapmetML, only: met_params
+!> open netcdf file
+    integer, intent(in) :: ncid
+!> timestep in file
+    integer, intent(in) :: timepos
+    integer :: start(7), count(7)
+
+    real(real64), allocatable :: normaliser(:,:)
+    real(real64), allocatable :: pdiff(:,:)
+    real(real64), allocatable :: cloud_water(:,:), cloud_ice(:,:)
+
+    integer :: ilevel, k, nr, i, j
+
+!.. get the correct ensemble/realization position, nr starting with 1, enspos starting with 0
+    nr = enspos + 1
+    if (enspos <= 0) nr = 1
+
+    allocate(pdiff(nx,ny))
+    allocate(normaliser(nx,ny))
+    allocate(cloud_water(nx,ny), cloud_ice(nx,ny))
+
+    precip3d(:,:,:) = 0.0
+    cw3d(:,:,:) = 0.0
+
+    normaliser(:,:) = 0.0
+
+    do k=nk,2,-1
+      ilevel = klevel(k)
+      call calc_2d_start_length(start, count, nx, ny, ilevel, &
+          enspos, timepos, met_params%has_dummy_dim)
+
+      pdiff(:,:) = 100*( (ahalf(k-1) - ahalf(k)) + (bhalf(k-1) - bhalf(k))*ps2 )
+      call nfcheckload(ncid, met_params%mass_fraction_cloud_condensed_water_in_air, start, count, cloud_water(:,:))
+      call nfcheckload(ncid, met_params%mass_fraction_cloud_ice_in_air, start, count, cloud_ice(:,:))
+
+      cw3d(:,:,k) = (abs(cloud_water(:,:)) + abs(cloud_ice(:,:))) * pdiff / g
+
+      ! Use cloud water to assign precipitation at model levels
+      normaliser(:,:) = normaliser + cw3d(:,:,k)
+      precip3d(:,:,k) = precip * cw3d(:,:,k)
+
+      call nfcheckload(ncid, met_params%cloud_fraction, start, count, cloud_cover(:,:,k))
+    enddo
+
+    block
+    use snapgrdML, only: ivlevel
+    integer :: klimit
+    klimit = ivlevel(nint(0.67*10000.0))
+    do k=nk,2,-1
+      do j=1,ny
+        do i=1,nx
+          if (normaliser(i,j) > 0.0) then
+            precip3d(i,j,k) = precip3d(i,j,k) / normaliser(i,j)
+          elseif (k == klimit) then
+            ! Put all precip at level closest to 0.67, analoguous
+            ! with the old formulation of the precipitation
+            precip3d(i,j,k) = precip(i,j)
+          endif
+        enddo
+      enddo
+    enddo
+    end block
+
+    ! block ! Debug
+    !   use snapfldML, only: garea
```

Comment:
Is this dead code, or do you want to keep it?

Reply:
Removed